relay-general/src/pii/processor.rs

```diff
@@ -195,6 +198,85 @@ impl<'a> Processor for PiiProcessor<'a> {
         replay.process_child_values(self, state)?;
         Ok(())
     }
+
+    fn process_event(
+        &mut self,
+        event: &mut Event,
+        _meta: &mut Meta,
+        state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        scrub_graphql(event);
+
+        event.process_child_values(self, state)?;
+
+        Ok(())
+    }
+}
+
+/// Scrubs GraphQL variables from the event.
+fn scrub_graphql(event: &mut Event) {
+    let mut keys: BTreeSet<&str> = BTreeSet::new();
+
+    let mut is_graphql = false;
+
+    // collect the variables keys and scrub them out.
+    if let Some(request) = event.request.value_mut() {
+        if let Some(Value::Object(data)) = request.data.value_mut() {
+            if let Some(api_target) = request.api_target.value() {
+                if api_target.eq_ignore_ascii_case("graphql") {
+                    is_graphql = true;
+                }
+            }
+
+            if is_graphql {
+                if let Some(Annotated(Some(Value::Object(variables)), _)) =
+                    data.get_mut("variables")
+                {
+                    for (key, value) in variables.iter_mut() {
+                        keys.insert(key);
+                        value.set_value(Some(Value::String("[Filtered]".to_string())));
+                    }
+                }
+            }
+        }
+    }
+
+    // scrub PII from the data object if they match the variables keys.
+    if let Some(contexts) = event.contexts.value_mut() {
+        if let Some(Context::Response(response)) = contexts.get_context_mut("response") {
+            if let Some(Value::Object(data)) = response.data.value_mut() {
+                if is_graphql {
```

Comment:
Can we move this out to, maybe, line `243` and have an early return if the `is_graphql` is `false`.  This way you will have fewer nested functions and will communicate clearly that if it's not `graphql` we do not care

Reply:
Good point, makes sense.